### PR TITLE
Add user controller

### PR DESCRIPTION
**Pull request by Redvin-dt**

### CODE REVIEW

File: server/src/main/java/ru/hse/server/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    UserController(UserService userService){
+        this.userService = userService;
+    }
+
+    @PostMapping("/registration")
+    public ResponseEntity postUser(@RequestBody UserEntity user) {
+        try {
+            userService.registration(user);
+            return ResponseEntity.ok("Пользователь сохранен");
```
Comment (devcrocod) [inline server/src/main/java/ru/hse/server/controller/UserController.java]: english pls

Comment (devcrocod) [reply server/src/main/java/ru/hse/server/controller/UserController.java]: Also consider using logger

File: server/src/main/java/ru/hse/server/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        } catch (Exception e) {
```
Comment (devcrocod) [inline server/src/main/java/ru/hse/server/controller/UserController.java]: If possible, it is better to catch several exceptions

File: server/src/main/java/ru/hse/server/entity/UserEntity.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+@Entity(name = "user")
+public class UserEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String login;
+    private String password;
+
+    public void setId(Long id) { // TODO: mb remove
```
Comment (devcrocod) [inline server/src/main/java/ru/hse/server/entity/UserEntity.java]: It’s probably really better to do final

File: server/src/main/java/ru/hse/server/entity/UserEntity.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        this.id = id;
+    }
+
+    public Long getId() {
```
Comment (devcrocod) [inline server/src/main/java/ru/hse/server/entity/UserEntity.java]: also you can use Lombok

File: server/src/main/java/ru/hse/server/exception/UserAlreadyExistException.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package ru.hse.server.exception;
+
+public class UserAlreadyExistException extends RuntimeException {
```
Comment (devcrocod) [inline server/src/main/java/ru/hse/server/exception/UserAlreadyExistException.java]: Doesn't spring have a corresponding exception?

And perhaps it’s better not to do it specifically for the user, but to do something more general: `EntityAlreadyExistException`.  And describe a specific ('user', `lesson`, `group`) exception message

File: server/src/main/java/ru/hse/server/exception/UserAlreadyExistException.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+    public UserAlreadyExistException(String message){
```
Comment (devcrocod) [inline server/src/main/java/ru/hse/server/exception/UserAlreadyExistException.java]: formatting

just to set checkbox "checking formatting before commit" in your ide

**Review by devcrocod — APPROVED** [review]
before merging please look at my comments

**Review by YuriiFilatov — APPROVED** [review]

**Pull request merged**